sklearn/metrics/cluster/unsupervised.py

```diff
@@ -350,7 +350,7 @@ def davies_bouldin_score(X, labels):
     intra_dists = np.zeros(n_labels)
     centroids = np.zeros((n_labels, len(X[0])), dtype=np.float)
     for k in range(n_labels):
-        cluster_k = safe_indexing(X, labels == k)
+        cluster_k = safe_indexing(X, np.flatnonzero(labels == k))
```

Comment:
we're still supporting bool, right? at least according to the docs?

Reply:
Nop. This was the original doc: https://scikit-learn.org/stable/modules/generated/sklearn.utils.safe_indexing.html

It was what @jnothman mentioned that `safe_indexing` was designed for CV splitter which required integer indices and was not supposed to work with boolean. Extended boolean support (which was my initial idea) would require to add some checking in the CV splitter.